src/crawlee/_autoscaling/snapshotter.py

```diff
@@ -25,7 +22,25 @@
 
 logger = getLogger(__name__)
 
-T = TypeVar('T')
+T = TypeVar('T', bound=Snapshot)
+
+
+class SortedSnapshotList(list[T]):
+    """A list that maintains sorted order by `created_at` attribute for snapshot objects."""
+
+    def add(self, item: T) -> None:
```

Comment:
nit - you  could use the [bisect](https://docs.python.org/3/library/bisect.html) stdlib module instead of writing your own binary search